autotest/gcore/vrt_read.py

```diff
@@ -1540,6 +1540,45 @@ def test_vrt_protocol():
     assert ds.GetRasterBand(1).XSize == 18
     assert ds.GetRasterBand(1).YSize == 16
 
+    with gdal.quiet_errors():
+        ds = gdal.Open("vrt://data/float32.tif?tr=120")
+        assert ds is None
+
+    ds = gdal.Open("vrt://data/float32.tif?tr=120,240")
+
+    assert ds.GetGeoTransform()[0] == 440720.0
+    assert ds.GetGeoTransform()[1] == 120.0
+    assert ds.GetGeoTransform()[5] == -240.0
+    assert ds.GetRasterBand(1).XSize == 10
+    assert ds.GetRasterBand(1).YSize == 5
+
+    ds = gdal.Open("vrt://data/float32.tif?r=bilinear&tr=120,240")
+    assert ds.ReadRaster(0, 0, 1, 1) == bytearray(
+        b"?\xf4\x00C"
```

Comment:
cf
```python
>>> import struct
>>> struct.unpack('f', b"?\xf4\x00C")[0]
128.95408630371094
```

```suggestion
    assert struct.unpack("f", ds.ReadRaster(0, 0, 1, 1)) == pytest.approx(
        128.95408630371094)
```

and similarly below